src/lib/caching/structures/User.ts

```diff
@@ -138,6 +138,34 @@ export class User<T = Client> extends Structure<T> {
 		return existing ? existing.delete() : Promise.resolve(null);
 	}
 
+	/**
+	 * Returns the users avatar url.
+	 * @param options The image size, format and other options.
+	 */
+	public avatarURL(options?: ImageURLOptions): string | null {
+		if (!this.avatar || !(this.client instanceof Client)) return null;
+		return this.client.api.cdn.userAvatar(this.id, this.avatar, options);
+	}
+
+	/**
+	 * Returns the users avatar url or the default discord avatar url if they don't have a avatar.
+	 * @param options The image size, format and other options.
+	 */
+	public displayAvatarURL(options?: ImageURLOptions): T extends Client ? string : null {
```

Comment:
I..really dislike the triple casting that is done in these files... Can't we do a generic, like `displayAvatarURL<K = T extends Client ? string : null>(blahBlah: optiowons): K` and cast as K?